saleor/core/extensions/plugin.py

```diff
@@ -0,0 +1,84 @@
+from typing import TYPE_CHECKING, List, Union
+
+from django_countries.fields import Country
+from prices import Money, MoneyRange, TaxedMoney, TaxedMoneyRange
+
+from ..taxes import TaxType
+
+if TYPE_CHECKING:
+    from ...checkout.models import Checkout, CheckoutLine
+    from ...product.models import Product
+    from ...account.models import Address
+    from ...order.models import OrderLine, Order
+
+
+class BasePlugin:
+    """Abstract class for storing all methods available for any plugin."""
+
+    def calculate_checkout_total(
+        self, checkout: "Checkout", discounts: List["DiscountInfo"]
+    ) -> TaxedMoney:
+        raise NotImplementedError()
+
+    def calculate_checkout_subtotal(
+        self, checkout: "Checkout", discounts: List["DiscountInfo"]
+    ) -> TaxedMoney:
+        raise NotImplementedError()
+
+    def calculate_checkout_shipping(
+        self, checkout: "Checkout", discounts: List["DiscountInfo"]
+    ) -> TaxedMoney:
+        raise NotImplementedError()
+
+    def calculate_order_shipping(self, order: "Order") -> TaxedMoney:
+        raise NotImplementedError()
+
+    def apply_taxes_to_shipping(
+        self, price: Money, shipping_address: "Address"
+    ) -> TaxedMoney:
+        raise NotImplementedError()
+
+    def get_tax_rate_type_choices(self) -> List[TaxType]:
+        raise NotImplementedError()
+
+    def calculate_checkout_line_total(
+        self, checkout_line: "CheckoutLine", discounts: List["DiscountInfo"]
+    ) -> TaxedMoney:
+        raise NotImplementedError()
+
+    def calculate_order_line_unit(self, order_line: "OrderLine") -> TaxedMoney:
+        raise NotImplementedError()
+
+    def apply_taxes_to_product(
+        self, product: "Product", price: Money, country: Country, **kwargs
+    ):
+        raise NotImplementedError()
+
+    def show_taxes_on_storefront(self) -> bool:
+        raise NotImplementedError()
+
+    def taxes_are_enabled(self) -> bool:
+        raise NotImplementedError()
+
+    def apply_taxes_to_shipping_price_range(
```

Comment:
I would group related methods together e.g. all "apply_taxes_" should be together, "calculate_checkout_" together, etc.